packages/wonder-blocks-popover/components/popover-anchor.js

```diff
@@ -0,0 +1,63 @@
+// @flow
+import * as React from "react";
+import * as ReactDOM from "react-dom";
+
+type Props = {|
+    /**
+     * Callback to be invoked when the anchored content is mounted.
+     * This provides a reference to the anchored content, which can then be
+     * used for calculating popover content positioning.
+     */
+    anchorRef: (?HTMLElement) => mixed,
+
+    /**
+     * The element that triggers the popover. This element will be used to
+     * position the popover. It can be either a Node or a function using the
+     * children-as-function pattern to pass an open function for use anywhere
+     * within children. The latter provides a lot of flexibility in terms of
+     * what actions may trigger the `Popover` to launch the
+     * [PopoverDialog](#PopoverDialog).
+     */
+    children: React.Element<any> | (({open: () => void}) => React.Node),
+
+    /**
+     * Called when the anchor is clicked
+     */
+    onClick: () => void,
+|};
+
+/**
+ * The element that triggers the popover dialog. This is also used as reference
+ * to position the dialog itself.
+ */
+export default class PopoverAnchor extends React.Component<Props> {
+    componentDidMount() {
+        const anchorNode = ((ReactDOM.findDOMNode(this): any): ?HTMLElement);
+
+        if (anchorNode) {
+            this.props.anchorRef(anchorNode);
+        }
+    }
+
+    render() {
+        const {children, onClick} = this.props;
+
+        if (typeof children === "function") {
+            return children({
+                open: onClick,
+            });
+        } else {
+            // add onClick handler to automatically open the dialog after
+            // clicking on this anchor element
+            return React.cloneElement(children, {
+                ...children.props,
+                onClick: children.props.onClick
+                    ? () => {
```

Comment:
Added this condition here in case the current Popover instance is adding another `onClick` handler to the anchor element. This happens b/c for this particular condition, I'm automatically adding an `onClick` handler to the `PopoverAnchor` component in order to open the `PopoverDialog` by clicking on this button.

e.g. This works for this condition:
```
<Popover
    {...props}
>
    <Button onClick={customClickHandler}>click to open PopoverDialog</Button>
</Popover>
```
In this case, if you click on the button, Popover will open the Dialog and will execute `customClickHandler` as well.

Reply:
Added a comment in line # 56